analysis/tools/sndSciFiTools.cxx

```
@@ -542,53 +543,35 @@ int snd::analysis_tools::showerInteractionWall(const TClonesArray &digiHits, int
    return showerInteractionWall(digiHits, selection_parameters, method, setup);
 }
 
-double computeMean(const std::vector<double>& values, double bin_width)
+double computeMean(const std::vector<double>& values)
 {
-    if (values.empty()) {
-        return 0.0;
-    }
-
-    double min_val = *std::min_element(values.begin(), values.end());
-    double max_val = *std::max_element(values.begin(), values.end());
-
-    double start = static_cast<double>(static_cast<int>(min_val - 1));
-    double end   = static_cast<double>(static_cast<int>(max_val + 1));
-
-    int nbins = static_cast<int>(std::ceil((end - start) / bin_width));
-    if (nbins <= 0) {
-        return 0.0;
+    if ( values.empty() ) 
+    {
+      return 0.0;
```

When the vector is empty, returning zero might be confusing, as the mean of an empty set doesn't really make sense. Might be better to raise an error in the calling code if there are no hits.